### DISCUSSION TITLE
(Too) long weather conditions on the taskbar.

### DISCUSSION THREAD

**Discussion by doterik** [valinet/ExplorerPatcher#897] [General]
Maybe you should shorten the weather condition on the taskbar.

For instance, just now I have 'Clear with periodic clouds', witch takes up a little more space than I want. This could perhaps be shortened with dots after a certain length ('Clear with period...'), and if you want to know the condition you only have to open the flyout.

Just a suggestion 😎

**Comment by valinet** [comment]
Then some will say the ellipsis looks bad... which kind of does. In 'Clear with period...', 'with period...' is actually useless since you can't tell what it refers to without opening the flyout (it could be 'periodic clouds', 'periodic showers', 'periodic thunders' etc), so imo it's actually worse. If I replace the messages, then I run into localization issues. Those are the messages that Google chose, indeed, they would better had chosen shorter ones...

A better solution is to use the "Icon and temperature", "icon only" or "Temperature only" display modes (i.e. hide the condition altogether).

**Reply by doterik** [reply]
Ok, you're the boss... although I think differently on almost every point you mention (except for localization issues, which are perhaps best solved by replacing whole words, possibly in the middle (maybe from a table)).

Dots or ellipses indicate to me that there is more information to find elsewhere, not entirely uncommon. For me, not completely useless.

If I turn off the condition, I have to open the flyout every time. A slightly shorter variant probably shows the entire condition in 9 cases out of 10, which I could put up with.

I think the difference is that I don't have such a wide screen that I can afford to lose more space on the taskbar.

Thanks anyway for your answer and a great app.

**Reply by valinet** [reply]
> except for localization issues, which are perhaps best solved by replacing whole words, possibly in the middle (maybe from a table)

It doesn't work, there's also the grammar. In Romanian adjectives are after nouns, in English they're before, just an example. Localization is a bitch, that's why it's best to let others do it and grab from what they have as much as possible.

> If I turn off the condition, I have to open the flyout every time. A slightly shorter variant probably shows the entire condition in 9 cases out of 10, which I could put up with.

You can also hover the mouse over it and the tooltip will be shown, which contains the condition when you chose to hide it.

As for the rest, I don't know what's not to agree on. 'Clear with period...' is as useful as 'Clear...', it doesn't offer more information than that, so if space is an issue, better to go with 'Clear...'.

Anyway, I have implemented an option that is basically the same as enlarge/shrink automatically, only it is allowed to go up to a certain (hardcoded) size. I guess that should do:

![image](https://user-images.githubusercontent.com/6503598/154977609-cedeafec-f0ca-4075-a103-c1cadbec3dba.png)

While I was at it, I made it so that when the condition is ellipsed, the tooltip shows it as well.
